File: 145_BinaryTreePostorderTraversal/python/BTPostOrder.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def postorderTraversal(self, root: TreeNode) -> List[int]:
        ans = []
        if None==root:
            return ans
        self.recrusivePostOrder(root, ans)
        return ans
    
    def recrusivePostOrder(self, node: TreeNode, store: list):
        if node.left != None:
            self.recrusivePostOrder(node.left, store)
        if node.right != None:
            self.recrusivePostOrder(node.right, store)
            
        store.append(node.val)

File: 145_BinaryTreePostorderTraversal/c#/BTPostOrder.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public IList<int> PostorderTraversal(TreeNode root) {
        List<int> ans = new List<int>();
        if(root==null){
            return ans;
        }
        recrusivePostOrder(root, ans);
        return ans;
    }
    
    private void recrusivePostOrder(TreeNode node, List<int> store){
        if(node.left!=null){
            recrusivePostOrder(node.left, store);
        }
        if(node.right!=null){
            recrusivePostOrder(node.right, store);
        }
        store.Add(node.val);
    }
}